core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -362,6 +362,16 @@ func (eb *EthBroadcaster) processUnstartedEthTxs(ctx context.Context, fromAddres
 	if err != nil {
 		return errors.Wrap(err, "processUnstartedEthTxs failed on handleAnyInProgressEthTx"), retryable
 	}
+	// Check if there are any unconfirmed txs that exceed a configurable threshold and log critical.
+	oldestUnconfirmedTx, err := eb.orm.FindOldestUnconfirmedTransactionFromAddress(fromAddress, eb.chainID)
```

Comment:
This introduces an extra additional query on every run. Can we not use the EthConfirmer to handle this, or the EthResender? Those already do a query for unconfirmed txes so we should be able to use that without issuing an extra one.

Reply:
Yes, I would assume you refer to [this](https://github.com/smartcontractkit/chainlink/blob/2587fca1cf5be0ae97257028913b451abc0802ec/core/chains/evm/txmgr/eth_broadcaster.go#L363) and [this](https://github.com/smartcontractkit/chainlink/blob/2587fca1cf5be0ae97257028913b451abc0802ec/core/chains/evm/txmgr/eth_broadcaster.go#L368), but these two only return a count instead of the actual tx, so we would have to change the existing queries. Any other specific place you have in mind for this check to be placed?

Reply:
@dimriou : I think you can add this logic here: https://github.com/smartcontractkit/chainlink/blob/dffc9f082516b2d5925782a7f6df12987f72b34c/core/chains/evm/txmgr/eth_resender.go#L111

The ORM call that we are making here, already returns all unconfirmed attempts for a particular Address.
You can go through this list, collect attempts that were broadcasted before a target block number, and use that data to log the errors.
This way, no new queries are issued to the ORM.
One thing I see, is that we may not be preserving the first broadcast timestamp. We only preserve BroadcastBeforeBlockNum, so it is hard to maybe find the time difference.
One thing you could do, is change the config value to depend on block numbers rather than timestamp, for the threshold.

Reply:
My initial thought was to add this to the `eth_resender`, but then I saw that the default resend interval is 5 seconds, and I thought it might end up spamming the logs.

Reply:
Also if we implement it on top of `FindEthTxAttemptsRequiringResend` with a block number variable instead,  we will create a weird correlation between `EthTxResendAfterThreshold` and the block number variable: the variable will always have to point to a block created before the `EthTxResendAfterThreshold`, otherwise the check will never return valid attempts.

Reply:
I think to do that we would have to keep separate timestamps for the different keys since the `FindEthTxAttemptsRequiringResend` method is called for each key. That's a bit of an overkill compared to the benefits. I suggest we log in the current interval we have, since there is already an initial 5 min cap, and see in practice if the logs will indeed be flooded.